drivers/input/keyboard/keyboard_gpio.c

```diff
@@ -0,0 +1,421 @@
+/*
+ * Copyright (c) 2022, tangchunhui@coros.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT keyboard_gpio
```

Comment:
I still don't know why this isn't done as a kscan driver, and this input system doesn't have a mapping to consume a kscan driver and map to input system events.

Reply:
I don't think you understand enough of this input subsystem framework either.